asserts/hardware_identity.go

```diff
@@ -151,3 +159,94 @@ func checkStringIsPEM(data []byte) (crypto.PublicKey, error) {
 
 	return pubKey, nil
 }
+
+// VerifySignature checks the signature of a given nonce against the hardware id key.
+// It is used by the model service to verify the request-id.
+// It currently supports key with algorithms RSA, DSA, ECDSA, and ED25519.
+// The hash algorithm used is also specified as a parameter.
+func (h *HardwareIdentity) VerifyNonceSignature(nonce, signature []byte, hashAlg crypto.Hash) error {
+	// The New() function can panic if the hash algorithm is not supported but does
+	// not return an error. We recover from the panic to avoid crashing the program.
+	hash := hashAlg.New()
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("Recovered from panic:", r)
+		}
+	}()
```

Comment:
why this?

Reply:
but the defer is after the call to New so probably you need a test and would need to set err instead of printing from inside it but let me check something